### ISSUE TITLE
new check for man pages

### ISSUE THREAD

**Issue by DimStar77** [rpm-software-management/rpmlint#918]
rpmlint should warn/err on things like this:

> /usr/share/man/man1/libcbor.3.gz

i.e. the check should validate that the directory `/usr/share/man/man<n>` only contains files matching `*.<n>%{?ext_man}` (ext_man is defined to the compression scheme used, e.g. `.gz`

**Comment by DimStar77** [comment]
also: I don't think `/usr/share/man/man<n>` is supposed to contain further subdirectories; another trigger to yell at packagers

**Comment by DimStar77** [comment]
Seems we were a bit too strict with the new checks

e.g it is very common for perl modules to have a file called
topic.3pm.gz in /usr/share/man/man3 - which seems to be valid

this implies the folder name `man<N>` should contain files in the form `*.<N>*.gz`

(I've seen other suffixed than pm)

**Comment by jengelh** [comment]
Due to automake, it has been common on GNU/Linux to place .3pm into /usr/share/man/man3.
This is in conflict with the way things were traditionally done (e.g. Solaris) where the full suffix is used, e.g. .1m ../man1m rather than ../man1.
An inquiry on the automake mailing list is pending.

**Comment by tmzullinger** [comment]
This appears to have caused a regression.  Running rpmlint-2.4.0 against https://kojipkgs.fedoraproject.org//packages/shigofumi/0.9/5.fc37/x86_64/shigofumi-0.9-5.fc37.x86_64.rpm results in:

```
shigofumi.x86_64: W: no-manual-page-for-binary shigofumi
```

The package contains a matching man page.  This works in 2.3.0 or in 2.4.0 if I revert `FilesCheck.py` to 718e1eb9^.

```
$ rpm -qplv shigofumi-0.9-5.fc37.x86_64.rpm | grep -E ' /usr/(bin|share/man)' | grep -w shigofumi
-rwxr-xr-x    1 root     root                   191240 Jul 23 08:38 /usr/bin/shigofumi
-rw-r--r--    1 root     root                     3941 Jul 23 08:38 /usr/share/man/cs/man1/shigofumi.1.gz
-rw-r--r--    1 root     root                     3372 Aug 18  2015 /usr/share/man/man1/shigofumi.1.gz
```

Before I go digging, is the cause obvious to anyone here who worked on this change?  My first thought is that the translated man page is what the code may be tripping over.

**Comment by marxin** [comment]
Thanks for the report, apparently I screwed up the regex pattern, I've got a patch candidate.

**Comment by tmzullinger** [comment]
Thanks @marxin!  I was hoping it would be more obvious to someone familiar with the code.  I also hope it's an easy fix.  Much appreciated!
